papers/numba/gufunc/gufunc.rst

```diff
@@ -0,0 +1,123 @@
+Generalized Ufuncs
+------------------
+
+In a previous section, we saw the power of ufuncs for expressing functions on
+arrays as functions on scalars combined with broadcasting.  This covers a
+large number of situations, but the elements of the output array can only
+depend on one element of each input at time.  If multiple elements from a
+single input array need to be combined, some other approach is required.
+
+For concreteness, let's suppose we want to compute a weighted average of the
+elements in each row of a 2D array, producing a 1D array as output.  A pure
+NumPy approach to this problem would look like:
+
+.. code-block:: python
+
+    def np_weighted_avg(x, weights):
+    return (x * weights).sum(axis=1)
```

Comment:
There seems to be an indentation problem here, or perhaps it's just Github showing it wrong.